fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -1044,7 +1133,10 @@ def _init_param_attributes(self, p: Parameter) -> None:
             # If we plan to keep the FP32 parameters on CPU, then pinning
             # memory allows us to later use non-blocking transfers when moving
             # the FP32 param shard to compute_device.
-            p._fp32_shard = p._fp32_shard.pin_memory()
+            if not self.ssd_offload:
+                # We don't pin memory when using ssd_offload since that results in OOM when
```

Comment:
I don't see a change about not using non-blocking transfers when ssd offload is on. The comment on 1134 mentioned that pin_memory is needed for non-blocking transfer. So I suspect that something is not be tested properly here.

Reply:
You don't need to set the non_blocking argument if pin_memory is not used. AFAIK the non_blocking will default to the synchronization point and not yield a perf gain. I added comments for ssd_offload.
I've tested this pretty extensively on large models and they haven't failed. In fact i was only able to test it once I fixed this memory leak in pin_memory.